nginx_vts_exporter.go

```diff
@@ -175,6 +175,31 @@ func newCacheMetric(metricName string, docString string, labels []string) *prome
 	)
 }
 
+func groupUpstreamsByServer(upstreamList []Upstream) []Upstream {
+	groupedByServer := make([]Upstream, 0)
+	upstreamsMap := make(map[string]int)
+
+	for _, upstream := range upstreamList {
+		if index, exists := upstreamsMap[upstream.Server]; exists {
+			groupedByServer[index].RequestCounter += upstream.RequestCounter
+			groupedByServer[index].InBytes += upstream.InBytes
+			groupedByServer[index].OutBytes += upstream.OutBytes
+			groupedByServer[index].Responses.OneXx += upstream.Responses.OneXx
+			groupedByServer[index].Responses.TwoXx += upstream.Responses.TwoXx
+			groupedByServer[index].Responses.ThreeXx += upstream.Responses.ThreeXx
+			groupedByServer[index].Responses.FourXx += upstream.Responses.FourXx
+			groupedByServer[index].Responses.FiveXx += upstream.Responses.FiveXx
+			groupedByServer[index].RequestMsec = (groupedByServer[index].RequestMsec + upstream.RequestMsec) / 2
+			groupedByServer[index].ResponseMsec = (groupedByServer[index].ResponseMsec + upstream.ResponseMsec) / 2
```

Comment:
`upstream.ResponseMsec` seems to be computed as an average, but computing an average on top of averages cannot technically be considered an average. For example, I think for a single server in upstream with `ResponseMsec=42ms` the aggregated `ResponseMsec` would be `21ms` (`(0 + 42) / 2 = 21`), which is different from the expected `42ms`.

I'd convert the averages back to the sums of times (assuming that `RequestCounter` is the denominator of the averages), and then compute the aggregated average using the sums, something like this:
```
if (groupedByServer[index].RequestCounter + upstream.RequestCounter) > 0 {
	groupedByServer[index].RequestMsec = (groupedByServer[index].RequestMsec * groupedByServer[index].RequestCounter + upstream.RequestMsec * upstream.RequestCounter) / (groupedByServer[index].RequestCounter + upstream.RequestCounter)
	groupedByServer[index].ResponseMsec = (groupedByServer[index].ResponseMsec * groupedByServer[index].RequestCounter + upstream.ResponseMsec * upstream.RequestCounter) / (groupedByServer[index].RequestCounter + upstream.RequestCounter)
}
groupedByServer[index].RequestCounter += upstream.RequestCounter  // note that this increment has to happen *after* the averages
```

Despite this little issue -- thank you for the great patch!

(Note, I'm not a maintainer of this project, I just faced the same issue and have reused your helpful solution).